src/deserializer.rs

```diff
@@ -84,48 +92,48 @@ fn read_ring(reader: &mut impl Read) -> io::Result<Vec<Point>> {
     Ok(ring)
 }
 
-fn read_point(reader: &mut impl Read) -> io::Result<Point> {
-    let x = read_u16(reader)?;
-    let y = read_u16(reader)?;
-    Ok(Point { x, y })
+fn read_point(reader: &mut impl Read) -> MyResult<Point> {
+    Ok(Point {
+        x: read_u16(reader)?,
+        y: read_u16(reader)?,
+    })
 }
 
-fn read_u8(reader: &mut impl Read) -> io::Result<u8> {
+fn read_u8(reader: &mut impl Read) -> MyResult<u8> {
     let mut buf = [0; 1];
     reader.read_exact(&mut buf)?;
     Ok(u8::from_be_bytes(buf))
 }
 
-fn read_u16(reader: &mut impl Read) -> io::Result<u16> {
+fn read_u16(reader: &mut impl Read) -> MyResult<u16> {
     let mut buf = [0; 2];
     reader.read_exact(&mut buf)?;
     Ok(u16::from_be_bytes(buf))
 }
 
-fn read_u32(reader: &mut impl Read) -> io::Result<u32> {
+fn read_u32(reader: &mut impl Read) -> MyResult<u32> {
     let mut buf = [0; 4];
     reader.read_exact(&mut buf)?;
     Ok(u32::from_be_bytes(buf))
 }
 
-fn read_usize32(reader: &mut impl Read) -> io::Result<usize> {
-    usize::try_from(read_u32(reader)?).map_err(|e| io::Error::new(ErrorKind::Unsupported, e))
+fn read_usize32(reader: &mut impl Read) -> MyResult<usize> {
+    Ok(usize::try_from(read_u32(reader)?)?)
```

Comment:
This is magic.... how does Rust know how to wrap the TryFromIntError into the custom error type?
Is this a feature of thiserror or somehow a feature of Rust?

Reply:
Both.  The ? operator uses From/Into to convert between the error types.  Thiserror creates this tr

Reply:
Ah, wow...